### ISSUE TITLE
Bad heartbeat causes panic

### ISSUE THREAD

**Issue by maxgarvey** [gmallard/stompngo#23]
We had a service that was being disrupted by a panic on the heartbeating logic:

```
Heartbeat Send Failure: write tcp 172.22.0.88:61613: connection reset by peer
panic: non-positive interval for NewTicker

goroutine 351 [running]:
runtime.panic(0x7746c0, 0xc2091ce2f0)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
time.NewTicker(0xffffffffcde10e9d, 0x62082dc)
	/usr/local/go/src/pkg/time/tick.go:23 +0xd8
github.com/gmallard/stompngo.(*Connection).sendTicker(0xc2080ba500)
	/srv/teamcity/work/7d0e7d4c05ce72c/src/github.com/monsooncommerce/last-offer-event-storage-service/Godeps/_workspace/src/github.com/gmallard/stompngo/heartbeats.go:122 +0x7b
created by github.com/gmallard/stompngo.(*Connection).initializeHeartBeats
	/srv/teamcity/work/7d0e7d4c05ce72c/src/github.com/monsooncommerce/last-offer-event-storage-service/Godeps/_workspace/src/github.com/gmallard/stompngo/heartbeats.go:100 +0x9b4
```

the timing bounds set in the previous heartbeat were creating a negative duration which caused a panic.

I have submitted a pull request that solves this by limiting the outstanding heartbeats to 10, and refactoring the time interval logic:
https://github.com/gmallard/stompngo/pull/22

**Comment by gmallard** [comment]
Closed: f6f1061aba

**Comment by gmallard** [comment]
If you are still using the stompngo package please try the current v.1.0.2 tag.

Tag information:

b1f1d966dd7b4c6e07278089c658aef2fb44749d refs/tags/v1.0.2

Document that version number in any issue reports.

Thanks for the support.

Guy
